react-native-spokestack.d.ts

```diff
@@ -0,0 +1,327 @@
+     * Specifies which input processor to use.
+     */
+    input?: SpokestackInput;
+    /**
+     * Stages is an array of classes through which to pass input
+     * Only the wakeword stage is relevant in iOS
+     */
+    stages: SpokestackStage[];
+    properties?: {
+      /**
+       * Audio sampling rate, in Hz
+       */
+      "sample-rate"?: number;
+      /**
+       * Speech frame width, in ms
+       */
+      "frame-width"?: number;
+      /**
+       * Trace level for logs
+       * DEBUG, PERF, INFO, and NONE
+       */
+      "trace-level"?: TraceLevel;
+      /**
+       * File system path to the "filter" Tensorflow-Lite model,
+       * which is used to calculate a mel spectrogram frame from the linear STFT;
+       * its inputs should be shaped [fft-width], and its outputs [mel-width]
+       */
+      "wake-filter-path"?: string;
+      /**
+       * File system path to the "encode" Tensorflow-Lite model,
+       * which is used to perform each autoregressive step over the mel frames;
+       * its inputs should be shaped [mel-length, mel-width], and its outputs [encode-width],
+       * with an additional state input/output shaped [state-width]
+       */
+      "wake-detect-path"?: string;
+      /**
+       * File system path to the "detect" Tensorflow-Lite model;
+       * its inputs shoudld be shaped [encode-length, encode-width],
+       * and its outputs
+       */
+      "wake-encode-path"?: string;
+
+      /**
+       * iOS-only
+       * A comma-separated list of wakeword keywords
+       * Only necessary when not passing the filter, detect, and encode paths.
+       */
+      wakewords?: string;
+
+      // Android-only AcousticNoiseSuppressor
+      // -----------------------------------------------
+      /**
+       * Noise policy
+       */
+      "ans-policy"?: "mild" | "medium" | "aggressive" | "very-aggressive";
+
+      // Android-only AcousticGainControl
+      // -----------------------------------------------
+      /**
+       * target peak audio level, in -dBFS for example,
+       * to maintain a peak of -9dBFS, configure a value of 9
+       */
+      "agc-compression-gain-db"?: number;
+      /**
+       * dynamic range compression rate, in dB
+       */
+      "agc-target-level-dbfs"?: number;
+
+      // Android-only VoiceActivityDetector
+      // -----------------------------------------------
+      /**
+       * Detector mode
+       */
+      "vad-mode"?: "quality" | "low-bitrate" | "aggressive" | "very-aggressive";
+      /**
+       * Rising-edge detection run length, in ms; this value determines
+       * how many positive samples must be received to flip the detector to positive
+       */
+      "vad-rise-delay"?: number;
+      /**
+       * Falling-edge detection run length, in ms; this value determines
+       * how many negative samples must be received to flip the detector to negative
+       */
+      "vad-fall-delay"?: number;
+
+      // Android-only wakeword
```

Comment:
This is supported on iOS too.

Reply:
All of the WakewordTrigger options are on iOS? Or do you mean only `wake-active-min` and `wake-active-max`?

Reply:
@noelweichbrodt Just need an answer here before I call it done.

Reply:
Will merge and we can fix later if needed.